Packages/com.unity.inputsystem/InputSystem/Actions/InputBinding.cs

```diff
@@ -842,6 +842,13 @@ public bool Matches(InputBinding binding)
             return Matches(ref binding);
         }
 
+        internal bool MatchesMask(ref InputBinding binding)
+        {
+            if (path != null)
```

Comment:
The matches function matches first on name, then path, then others like actions and groups.
It seems if path is non null we are skipping those other match checks.

Is this valid - is only one of path,name,action of group set or do we actually need the fallback cases?

I.e. perhaps this should be 

    if (path != null)
    {
        if (InputControlPath.BindingPathMatchesMask(path, binding.path))
            return true;
    }
    return Matches(ref binding, MatchOptions.EmptyGroupMatchesAny);

Reply:
yes, true. I think till now the binding mask is only used with a path, but if it changes in future that will be a safer way

Reply:
These test seem to suggest it works with groups
Actions_CanRecordActions_AndReadTheDataEvenIfBindingsHaveChanged
Actions_CanMaskOutBindingsByBindingGroup_OnAction_WhenEnabled

action.bindingMask = new InputBinding {groups = "B"};

Reply:
I never saw this tests before. You're right.